CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Models/DrugDosingModel/DrugDosingModelSteppable.py

```diff
@@ -0,0 +1,215 @@
+                                                     y_scale_type='linear',
+                                                     grid=True,
+                                                     config_options={'legend': True})
+        colors = ['blue', 'red', 'green', 'yellow', 'white']
+        for c, var in zip(colors, self.ddm_vars):
+            self.ddm_data_win.add_plot(var, style='Dots', color=c, size=5)
+
+        self.rmax_data_win = self.add_new_plot_window(title='r_max vs Time',
+                                                      x_axis_title='Time (hours)',
+                                                      y_axis_title='r_max',
+                                                      x_scale_type='linear',
+                                                      y_scale_type='linear',
+                                                      grid=True,
+                                                      config_options={'legend': True})
+
+        self.rmax_data_win.add_plot('rmax', style='Dots', color='red', size=5)
+
+    def start(self):
+
+        # set model string
+        self.drug_model_string, self.ddm_vars = self.set_drug_model_string(Drug, Available1, Available2, Available3,
+                                                                           Available4,
+                                                                           k0, d0, k1, d1, k2, d2, k3, d3, d4,
+                                                                           first_dose,
+                                                                           initial_dose, dose_interval, dose)
+
+        # init sbml
+        self.add_free_floating_antimony(model_string=self.drug_model_string, step_size=days_2_mcs,
+                                        model_name='drug_dosing_model')
+
+        # init plots
+        if self.plot_ddm_data:
+            self.init_plots()
+
+        # init save data
+        if self.write_ddm_data:
+            from pathlib import Path
+            for key, rel_path in self.data_files.items():
+                self.data_files[key] = Path(self.output_dir).joinpath(rel_path)
+                with open(self.data_files[key], 'w'):
+                    pass
+
+        # Post reference to self
+        self.shared_steppable_vars[self.drug_dosing_model_key] = self
+
+    def get_rmax(self, avail4):
+        return (1 - nCoVUtils.hill_equation(avail4, rel_avail4_EC50 * self.max_avail4, 2)) * replicating_rate
+
+    def step(self, mcs):
+        rmax = self.get_rmax(self.sbml.drug_dosing_model['Available4'])
+        print(rmax)
+        for cell in self.cell_list_by_type(self.INFECTED, self.VIRUSRELEASING):
+            # print(self.vr_model_name)
+            # vr_model_name = self.vr_model_name
+            # todo: grab sbml from variable
+            # vr_model = cell.sbml.vr_model_name.__str__()
+            vr_model = cell.sbml.viralReplication
```

Comment:
Recommend making this steppable a subclass of `ViralInfectionVTMSteppableBasePy` and then using the class attribute `vr_model_name`. This would facilitate usage of this module with other viral replication models.

Reply:
Line 172 would become `vr_model = getattr(cell.sbml, self.vr_model_name)`